proposals/lip-0051.md

```diff
@@ -769,14 +770,245 @@ if the above send function does not fail:
+   * Accross all element of the `supplySubstore` array, all values given for `localID` must be unique.
+   * The `supplySubstore` array must be in ascending order of `localID`.
+   * Accross all elements of the `escrowSubstore` array, the pairs `escrowChainID, localID` must be unique (each pair appears at most once). 
+   * The `escrowSubstore` array must be in ascending order of `escrowChainID`. For a given `escrowChainID`, the entries must be in ascending order of `localID`.
+   * In the `terminatedEscrowSubstore` array, each element must be unique (appears at most once). 
+   * The `terminatedEscrowSubstore` array must be in ascending order.
+
+* For each entry `userEntry` in `genesisBlockAssetObject.userSubstore`, create an entry in the user substore  with
+  ```python
+  storeKey = userEntry.address || uint32be(userEntry.tokenID.chainID) || uint32be(userEntry.tokenID.localID)
+  storeValue = {"availableBalance": userEntry.availableBalance,
+                "lockedBalances": [{"moduleID": lockedBalance.moduleID,
+                                    "amount": lockedBalance.amount
+                                   } for each lockedBalance in userEntry.lockedBalances]
+               } serialized using userStoreSchema.
+  ```
+
+* For each entry `supplyEntry` in `genesisBlockAssetObject.supplySubstore`, create an entry in the supply substore  with
+  ```python
+  storeKey = uint32be(supplyEntry.localID)
+  storeValue = {"totalSupply": supplyEntry.totalSupply} serialized using supplyStoreSchema.
+  ```
+
+* For each entry `escrowEntry` in `genesisBlockAssetObject.escrowSubstore`, create an entry in the escrow substore  with
+  ```python
+  storeKey = uint32be(escrowEntry.escrowChainID) || uint32be(escrowEntry.localID)
+  storeValue = {"amount": escrowEntry.amount} serialized using escrowStoreSchema.
+  ```
+
+* Create an entry in the available local ID substore with
+  ```python
+  storeKey = EMPTY_BYTES
+  storeValue = {"nextAvailableLocalID": genesisBlockAssetObject.availableLocalIDSubstore.nextAvailableLocalID} 
+               serialized using availableLocalIDStoreSchema.
+  ```
+
+* For each entry `terminatedChainID` in `genesisBlockAssetObject.terminatedEscrowSubstore`, create an entry in the terminated escrow substore with
+  ```python
+  storeKey = uint32be(terminatedChainID)
+  storeValue = {"escrowTerminated": True} serialized using terminatedEscrowSchema.
+  ```
+  
+Once the module store is initialized, its validity is attested via the two checks below.
+* Check that for each native token the total supply is correct. 
+  This can be done by checking that the function below returns `True`.
+  ```python
+  validateSupplyStoreEntries():
+      computedSupplies = {}
+      for (storeKey, storeValue) in user substore:
+          chainID = storeKey[20:24] # part of the key corresponding to the chain ID of the token
+          localID = storeKey[20:28] # part of the key corresponding to the local ID of the token
```

Comment:
is the starting index of `20` here correct? it seems the same for `chainID` above

Reply:
nice catch, should be 24. fixed